### ISSUE TITLE
Closing a DatalakeFileClient (from get_file_client) makes subsequent DatalakeFileClient operations fail

### ISSUE THREAD

**Issue by ljos** [Azure/azure-sdk-for-python#14497]
- **Package Name**: azure-storage-file-datalake
- **Package Version**: 12.1.2
- **Operating System**: MacOS
- **Python Version**: python3.8

**Describe the bug**
Opening a FileSystemClient as a context manager and then getting multiple file clients, if the first one is closed, all subsequent DataLakeFileClient operations will fail.

**To Reproduce**
```
with FileSystemClient() as fs_client:
    with fs_client.get_file_client("file1.txt") as f_client:
        f_client.create_file()
    with fs_client.get_file_client("file2.txt") as f_client:
        f_client.create_file()
```

The second `with` will fail with 

```
AttributeError: 'NoneType' object has no attribute 'request'
```

on `azure/core/pipeline/transport/_requests_basic.py", line 259, in send`

This is because `self.session` was set to None as a result of exiting the first context manager.

**Expected behavior**
That opening and subsequently closing a DatalakeFileClient as a context manager does not affect other instances of file clients and that the above code works.

**Comment by tasherif-msft** [comment]
Hi @ljos, I was able to reproduce your issue and have been investigating it.
In the meantime a possible workaround for your example:
```
fs_client = FileSystemClient()
try: 
    fs_client.get_file_client("file1.txt").create_file()
    fs_client.get_file_client("file2.txt").create_file()
finally: 
    fs_client.close() 
```

@xiangyan99 @xiafu-msft It seems like this might be a core related issue?

**Comment by ljos** [comment]
This is exactly what I do, but I would prefer to close the files properly at the time I don't need them anymore.

**Comment by xiafu-msft** [comment]
Hi @ljos 
Thanks for finding this bug! we will fix it

**Comment by tasherif-msft** [comment]
Hi @ljos, I have looked into the bug and found the cause very interesting. The reason this happens is because when `f_client` is created `fs_client` shares its pipeline object. This object contains a `transport` which contains a `session`. This session is closed when the file client is released and therefore closed in the file system as well.
I have created a PR for this fix :)
 https://github.com/Azure/azure-sdk-for-python/pull/14910
